error[E0412]: cannot find type `Cake` in this scope

 --> mutant.rs:8:34

  |

8 | # [ rustc_variance ] const GOO : Cake = foo ( ) ;

  |                                  ^^^^ not found in this scope



error[E0425]: cannot find function `foo` in this scope

 --> mutant.rs:8:41

  |

8 | # [ rustc_variance ] const GOO : Cake = foo ( ) ;

  |                                         ^^^ not found in this scope



error[E0208]: [+, -, o, *]

 --> mutant.rs:3:1

  |

3 | / enum Base < 'a , 'b , 'c : 'b , 'd > {

4 | | Test8A ( extern "Rust" fn ( & 'a isize ) ) ,

5 | | Test8B ( & 'b [ isize ] ) ,

6 | | Test8C ( & 'b mut & 'c str ) ,

7 | | }

  | |_^



error: internal compiler error: compiler/rustc_typeck/src/variance/mod.rs:44:9: asked to compute variance for wrong kind of item

 --> mutant.rs:8:22

  |

8 | # [ rustc_variance ] const GOO : Cake = foo ( ) ;

  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^



thread 'rustc' panicked at 'Box<dyn Any>', /rustc/e8cb1a4a567ce88e459ffd431207eff2e0f0ffa5/compiler/rustc_errors/src/lib.rs:953:9

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace



note: the compiler unexpectedly panicked. this is a bug.



note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md



note: rustc 1.55.0-nightly (e8cb1a4a5 2021-06-27) running on x86_64-unknown-linux-gnu



note: compiler flags: -Z mir-opt-level=3 -Z new-llvm-pass-manager=yes -Z unsound-mir-opts=yes -Z polonius=yes -Z polymorphize=yes -C debuginfo=2 -C opt-level=3 --crate-type staticlib



query stack during panic:

#0 [variances_of] computing the variances of `GOO`

#1 [analysis] running analysis passes on this crate

end of query stack

error: aborting due to 4 previous errors



Some errors have detailed explanations: E0412, E0425.

For more information about an error, try `rustc --explain E0412`.

